pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -30,30 +30,33 @@ import (
 
 const primaryPodInterfaceName = "eth0"
 
-type VMNetworkConfigurator struct {
+type NetworkingConfigurator interface {
+	Setup() error
+}
+
+type InfraNetworkingConfigurator struct {
 	vmi          *v1.VirtualMachineInstance
 	handler      netdriver.NetworkHandler
 	cacheFactory cache.InterfaceCacheFactory
+	launcherPID  int
+	nicMaker     NicGenerator
 }
 
-func newVMNetworkConfiguratorWithHandlerAndCache(vmi *v1.VirtualMachineInstance, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
-	return &VMNetworkConfigurator{
-		vmi:          vmi,
-		handler:      handler,
-		cacheFactory: cacheFactory,
-	}
-}
-
-func NewVMNetworkConfigurator(vmi *v1.VirtualMachineInstance, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
-	return newVMNetworkConfiguratorWithHandlerAndCache(vmi, &netdriver.NetworkUtilsHandler{}, cacheFactory)
+type NetworkingSpecGenerator struct {
+	vmi          *v1.VirtualMachineInstance
+	handler      netdriver.NetworkHandler
+	cacheFactory cache.InterfaceCacheFactory
+	domXML       api.Domain
+	nicMaker     NicGenerator
 }
 
-func (v VMNetworkConfigurator) getNICs() ([]podNIC, error) {
-	return v.getNICsWithLauncherPID(nil)
-
+type NicGenerator struct {
+	vmi          *v1.VirtualMachineInstance
+	handler      netdriver.NetworkHandler
+	cacheFactory cache.InterfaceCacheFactory
 }
 
-func (v VMNetworkConfigurator) getNICsWithLauncherPID(launcherPID *int) ([]podNIC, error) {
```

Comment:
What I don't like is the use like `getNIC(nil)` we don't know what nil so if we do `getNICWithLauncherPID(nil)` is clear what nil is, another option is to do the opposite `getNICWithtouLauncherPID()` and this will call `getNIC(nil)`.

Reply:
If that is your issue, easily worked around:

```golang
launcherPid := nil
nic := getNIC(launcherPid)
...
```